### PR TITLE
Merge google -> main

### PR DESCRIPTION
**Pull request by ScottTodd**
During `google` -> `main` integrate:
* ab78546 Tidy up getIndexValue implementation.

From `google` branch:
* 7808740bf Synchronize submodules with LLVM at llvm/llvm-project@4f9217c5194b
* 1ad5cf67b Run bazel_to_cmake.
* b1866da22 Integrate LLVM at llvm/llvm-project@4f9217c5194b
* ea7e38a0e Integrate LLVM at llvm/llvm-project@f53fafbacbc2
* 5894e45dc Merge pull request #7008 from ScottTodd:main-to-google
* 4a55090f6 Synchronize submodules with LLVM at llvm/llvm-project@6c8ff4032e2b
* 5af7ea1f8 Integrate LLVM at llvm/llvm-project@6c8ff4032e2b
* 708976757 Integrate LLVM at llvm/llvm-project@9da62d3ed9d7
* c1320f9a0 Integrate LLVM at llvm/llvm-project@585c594d749a

### CODE REVIEW

File: iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp [hunk 1]
```diff
@@ -308,9 +309,9 @@ class HALDispatchABI {
 
   Value getIndexValue(Location loc, int64_t value, OpBuilder &builder) {
     SmallVector<Value> folded;
-    builder.createOrFold<UnrealizedConversionCastOp>(
+    builder.createOrFold<LLVM::ConstantOp>(
         folded, loc, typeConverter->convertType(builder.getIndexType()),
-        builder.createOrFold<ConstantIndexOp>(loc, value));
+        builder.getI64IntegerAttr(value));
     return folded.front();
```
Comment (ScottTodd) [inline iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp:315]: errr... why does `getIndexValue` no longer return an index?

File: iree/compiler/Codegen/SPIRV/test/tile_and_vectorize.mlir [hunk 2]
```diff
@@ -26,11 +26,7 @@ hal.executable private @matmul  {
       workgroup_size = [16: index, 8: index, 1: index],
       translation.info = {passPipeline = 6 : i32, workloadPerWorkgroup = [8, 16]}
     }
-    builtin.module attributes {
-      spv.target_env =
-        #spv.target_env<#spv.vce<v1.3, [Shader], [SPV_KHR_storage_buffer_storage_class]>,
-                        {max_compute_workgroup_invocations = 128 : i32,
-                         max_compute_workgroup_size = dense<[128, 128, 64]> : vector<3xi32>}>} {
+    builtin.module {
```
Comment (ScottTodd) [inline iree/compiler/Codegen/SPIRV/test/tile_and_vectorize.mlir:29]: are these attributes okay to remove from the test?

Comment (MaheshRavishankar) [reply iree/compiler/Codegen/SPIRV/test/tile_and_vectorize.mlir:29]: This is fine. They are moved to `hal.executable.variant` op.

Comment (antiagainst) [reply iree/compiler/Codegen/SPIRV/test/tile_and_vectorize.mlir:29]: Yes. We right now don't rely on such information anymore after the pipeline refactoring. Actually it's a good thing to clean them up. Let me do a pass afterwards.

Comment (antiagainst) [reply iree/compiler/Codegen/SPIRV/test/tile_and_vectorize.mlir:29]: As more context, now the pipeline attaches the configuration as the `#config` at L11 to drive CodeGen. `#config` is typically attached by an earlier pass. But here this test does not involve that pass (`SPIRVLowerExecutableTargetPass`).

File: iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp (hunk 1 above)
Comment (MaheshRavishankar) [reply iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp:315]: Looking at the lit tests, the op is already returning an i64. Are we assuming all devices are i64 for index types. If so this should be fine

Comment (benvanik) [reply iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp:315]: we aren't - riscv32 isn't, for example - needs to be index

Comment (ScottTodd) [reply iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp]: discussed [on Discord](https://discord.com/channels/689900678990135345/690274711523164166/885943515304054824), seems fine, just looked weird

**Review by GMNGeoffrey — APPROVED** [review]
I have reservations about how LLVM lowerings are treating `unrealized_conversion_cast`, but I think it's not actually getting worse in IREE with this patch. Just alerted me to what's going on :-)

**Pull request merged**